plugins/utils/alpha.nix

```diff
@@ -139,9 +146,24 @@ in
     let
       layoutDefined = cfg.layout != [ ];
       themeDefined = cfg.theme != null;
+
+      opt = options.plugins.alpha;
     in
     mkIf cfg.enable {
-      extraPlugins = [ cfg.package ] ++ (optional cfg.iconsEnabled pkgs.vimPlugins.nvim-web-devicons);
+      # TODO: deprecated 2024-08-29 remove after 24.11
+      warnings = lib.mkIf opt.iconsEnabled.isDefined [
+        ''
+          nixvim (plugins.alpha):
+          The option definition `plugins.alpha.iconsEnabled' in ${showFiles opt.iconsEnabled.files} has been deprecated; please remove it.
+          You should use `plugins.alpha.iconsPackage' instead.
+        ''
+      ];
```

Comment:
This does not trigger due to the `default` value? It's freeform types that are strange in that way then?

Reply:
I removed the default for it so that it only triggers when someone has explicitly declared it in their configuration.

Reply:
Sorry I missed that in the diff --'